File: Ressurser/New_Steiner_Algo/mst_print.m
function [] = mst_print(V)
% function [] = mst_print(V)
% print the minimum-spanning-tree
%
% Reggie Davidrajuh (c) August 2011

TotalWT = 0;
MST = V.MST;

disp(' ');
disp('Printing the edges that are scanned to make MST ... ');
% now print complete MST
for i=1:length(V.nodes)-1
    mst = MST(i, :);
    u = mst(1); v = mst(2); wt = mst(3);
    TotalWT = TotalWT + wt;
    
    disp(['Edge-', int2str(i), ':  ', V.nodes(v).name, '-',...
        V.nodes(u).name, '    Wt: ', int2str(wt)]);
end

% finally, display the total weight
disp(['Total Weight : ', int2str(TotalWT)]);   

File: Kode/mst_print.py
import numpy as np;
from sample_graph import SampleGraph;

def mst_print(MST):
    print("Printing the MST with the vertices and edges:")
    TotalWT = 0
    for i in range(len(MST.Nodes)-1):
        mst = MST.A[:][i]
        u = mst[0]
        v = mst[1]
        wt = mst[2]
        TotalWT += wt
        print(f"Edge-{i}: {MST.Nodes[u]}-{MST.Nodes[v]} Wt:{wt}")
    print(f"Total Weight: {TotalWT}")
